### PR TITLE
chore: removed console warnings

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Description

Ticket: [ET-6](https://hpe-aiatscale.atlassian.net/browse/ET-6)

A few code changes intending to remove console warnings.

### Regression check

- make sure that all components still works in the intended way.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[ET-6]: https://hpe-aiatscale.atlassian.net/browse/ET-6?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by keita-determined** [comment]
could you make a pr in determined repo using this change?

File: src/kit/ResponsiveGroup.tsx [hunk 1]
```diff
@@ -55,11 +55,12 @@ const ResponsiveGroup: React.FC<Props> = ({
...
-    setNumVisible((prev) => {
-      if (prev !== visible) onChange?.(visible);
-      return visible;
-    });
-  }, [childrenArray, gap, maxVisible, onChange, size.width]);
+
+    if (visible !== numVisible) {
+      onChange?.(visible);
+      setNumVisible(visible);
+    }
```
Comment (EmilyBonar) [inline src/kit/ResponsiveGroup.tsx:62]: nit: this does get rid of the error message, but I worry that using both `numVisible` and `setNumVisible` in the same effect will cause it to be run too often, even if it's not an endless loop. see if something like this works:
```
let visibilityChanged = false;
setNumVisible((prevVisibility) => {
   if (prevVisibility !== visible) visibilityChanged = true;
   return visible;
}
if (visibilityChanged) onChange?.(visible)
```

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
I'm still seeing the following error in the console when I run the design kit page locally:
```
Warning: Invalid DOM property `fill-rule`. Did you mean `fillRule`?
svg
IconBase2@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:7287:14
span
Icon<@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:7343:19
CloseCircleFilled
IconNode@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:8427:7
div
DomWrapper4@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:7471:20
genCSSMotion/CSSMotion<@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:7859:26
Alert@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:8476:7
Alert
div
div
div
ElevationWrapper@http://localhost:5173/src/kit/internal/Elevation.tsx:14:33
Surface@http://localhost:5173/src/kit/Surface.tsx:15:17
SurfaceCard@http://localhost:5173/src/DesignKit.tsx?t=1709924343970:156:21
article
ComponentSection@http://localhost:5173/src/DesignKit.tsx?t=1709924343970:136:26
AlertSection
main
div
ElevationWrapper@http://localhost:5173/src/kit/internal/Elevation.tsx:14:33
div
div
Spin@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:52734:7
SpinFC@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:52784:7
div
Spinner@http://localhost:5173/src/kit/Spinner.tsx:17:17
div
App@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:72908:7
ProviderChildren@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:6653:7
LocaleReceiver@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:5490:7
ConfigProvider
div
UI@http://localhost:5173/src/kit/Theme/UI.tsx:109:19
UIProvider@http://localhost:5173/src/kit/Theme/UI.tsx:42:27
DesignKit@http://localhost:5173/src/DesignKit.tsx?t=1709924343970:15497:19
div
App@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:72908:7
ConfirmationProvider@http://localhost:5173/src/kit/useConfirm.tsx?t=1709924343966:47:37
div
App@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:72908:7
ProviderChildren@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:6653:7
LocaleReceiver@http://localhost:5173/node_modules/.vite/deps/antd.js?v=f5c79a6a:5490:7
ConfigProvider
div
UI@http://localhost:5173/src/kit/Theme/UI.tsx:109:19
UIProvider@http://localhost:5173/src/kit/Theme/UI.tsx:42:27
DesignKitContainer@http://localhost:5173/src/DesignKit.tsx?t=1709924343970:15678:37 react-dom.development.js:86:29
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -37,7 +37,7 @@ commands:
       - save_cache:
           key: deps-{{ checksum "/tmp/get-deps.cachekey" }}
           paths:
-            - "node_modules"
+            - 'node_modules'
```
Comment (keita-determined) [inline .circleci/config.yml:40]: whats this change for?

File: src/kit/Message.tsx [hunk 3]
```diff
@@ -38,7 +38,7 @@ const Message: React.FC<Props> = ({ action, description, title, icon }: Props) =
     <div className={classes.join(' ')}>
       {icon && getIcon(icon)}
       {title && <Title>{title}</Title>}
-      {description && <p className={css.description}>{description}</p>}
+      {description && <span className={css.description}>{description}</span>}
```
Comment (keita-determined) [inline src/kit/Message.tsx:41]: whats this for? this might change UI?

File: src/kit/ResponsiveGroup.tsx [hunk 4]
```diff
@@ -32,7 +32,7 @@ const ResponsiveGroup: React.FC<Props> = ({
   maxVisible = 3,
   onChange,
 }: Props) => {
-  const [, setNumVisible] = useState<number>();
+  const [numVisible, setNumVisible] = useState<number>();
```
Comment (keita-determined) [inline src/kit/ResponsiveGroup.tsx:35]: this state should be initialized

**Review by keita-determined — CHANGES_REQUESTED** [review]

**Review by keita-determined — CHANGES_REQUESTED** [review]
> It comes from the icon version of the`Alert` component, which comes from `antd` 😅

isnt it fixable? i dont see the warning [here](https://codesandbox.io/p/sandbox/icon-antd-5-15-1-forked-xlk2jx?file=%2Fpackage.json%3A4%2C28) 

---

`console.js:213 Warning: [antd: Form.Item] `defaultValue` will not work on controlled Field. You should use `initialValues` of Form instead. ` after opening modals.

---

Could you make a PR in determined too along with the warning fixes in determined repo and this fix in hew to see if everything works?

File: src/kit/ResponsiveGroup.tsx [hunk 5]
```diff
@@ -55,12 +55,18 @@ const ResponsiveGroup: React.FC<Props> = ({
       },
       { accumulatedWidth: 0, visible: 0 },
     );
+
     setNumVisible((prev) => {
-      if (prev !== visible) onChange?.(visible);
-      return visible;
+      if (prev !== visible) return visible;
+
+      return prev;
```
Comment (ashtonG) [inline src/kit/ResponsiveGroup.tsx:62]: i don't think you need a functional update for this value anymore -- these are numbers, so `Object.is` will return true if you pass the same value in, no?

File: src/kit/LineChart.tsx [hunk 6]
```diff
@@ -51,7 +51,7 @@ const getScientificNotationTickValues: uPlot.Axis['values'] = (_self, rawValues)
 interface ChartProps {
   focusedSeries?: number;
   height?: number;
-  key?: Options['key'];
+  optionKey?: Options['key'];
```
[the remark this answers is not in the document]
Comment (EmilyBonar) [reply src/kit/LineChart.tsx:54]: Yeah I got that, just making a note for when you merge

File: src/kit/ResponsiveGroup.tsx (hunk 5 above)
Comment (EmilyBonar) [reply src/kit/ResponsiveGroup.tsx:62]: Yeah, I think you should be able to just do `setNumVisible(visible)`

**Review by keita-determined — COMMENTED** [review]
> console.js:213 Warning: [antd: Form.Item] defaultValuewill not work on controlled Field. You should useinitialValuesof Form instead. after opening modals.

this warning is not fixed yet. 

> I think that this is only possible after the merge of this one, no?!

you can do that without merging it. [ref](https://github.com/determined-ai/determined/pull/8862/commits/b3c6bdfac23471cf0221873a6040363e8a31b3a1)

**Review by EmilyBonar — APPROVED** [review]
Looks good, we can address that last warning later

**Pull request merged**